othello_kotlin/app/src/main/kotlin/othello/Board.kt

```diff
@@ -0,0 +1,218 @@
+                if (getSquare(pos) != color) {
+                    continue
+                }
+                value += numSteps
+                directions.add(step)
+            }
+            if (value > 0) {
+                moves.add(Move(square, value, color, directions))
+            }
+        }
+        return moves.sortedWith(
+            compareByDescending<Move> { it.value }.thenBy { it.square.x }
+                .thenBy { it.square.y },
+        )
+    }
+
+    /** Print board with available move coordinates and the resulting points gained.*/
+    fun printMoves(moves: Collection<Move>) {
+        printColor("  Possible moves (${moves.size}):", Color.YELLOW)
+        // Convert board from Disk enums to strings
+        val formattedBoard = ArrayList<String>(board.size)
+        formattedBoard.addAll(board.map { it.boardChar() })
+        // Add possible moves to board
+        for (move in moves) {
+            println("  $move")
+            val (x, y) = move.square
+            formattedBoard[y * size + x] = getColor(move.value.toString(), Color.YELLOW)
+        }
+        // print board with move positions
+        print("   ")
+        for (i in indices) {
+            print(" $i")
+        }
+        for (y in indices) {
+            print("\n  $y")
+            for (x in indices) {
+                print(" ${formattedBoard[y * size + x]}")
+            }
+        }
+        println("")
+    }
+
+    /** Print current score for both players.*/
+    fun printScore() {
+        val (black, white) = playerScores()
+        println("\n$this")
+        println(
+            "Score: ${getColor(black.toString(), Disk.Black.diskColor())} | " +
+                getColor(white.toString(), Disk.White.diskColor()),
+        )
+    }
+
+    /** Calculates the final score and returns the winning player.*/
+    fun result(): Disk {
+        val sum = score()
+        return when {
+            sum < 0 -> Disk.Black
+            sum > 0 -> Disk.White
+            else -> Disk.Empty
+        }
+    }
+
+    /** Check that the given coordinates are inside the board.*/
+    private fun checkCoordinates(x: Int, y: Int): Boolean {
+        return x in 0 until size && y in 0 until size
+    }
+
+    /** Returns the state of the board (empty, white, black) at the given coordinates.*/
+    private fun getSquare(square: Square): Disk? {
+        val (x, y) = square
+        return if (checkCoordinates(x, y)) {
+            board[y * size + x]
+        } else {
+            null
+        }
+    }
+
+    /** Count and return the number of black and white disks. */
+    private fun playerScores(): Pair<Int, Int> {
+        var black = 0
+        var white = 0
+        for (disk in board) {
+            when (disk) {
+                Disk.White -> ++white
+                Disk.Black -> ++black
+                Disk.Empty -> {}
+            }
+        }
+
+        return black to white
```

Comment:
would prefer to use `Pair(black, white)` here instead, I usually use `to` operator only with maps, e.g.

`mapOf("key" to "some value")`